### PR TITLE
chore: remove remainder of native api

### PR DESCRIPTION
**Pull request by rb-determined-ai**
The Native API, deprecated in 0.13.5 (September 2020) is now completely
removed.

### CODE REVIEW

**Comment by azhou-determined** [comment]
do we need to remove the det.experimental native usage from user-facing docs?

File: harness/determined/_experiment_config.py [hunk 1]
```diff
@@ -8,9 +8,6 @@ def debug_enabled(self) -> bool:
     def scheduling_unit(self) -> int:
         return int(self.get("scheduling_unit", 100))
 
-    def native_enabled(self) -> bool:
-        return "internal" in self and self["internal"] is not None and "native" in self["internal"]
-
     def native_parallel_enabled(self) -> bool:
```
Comment (azhou-determined) [inline harness/determined/_experiment_config.py:11]: completely unrelated, but what even is this?

**Review by azhou-determined — APPROVED** [review]
🥳

File: harness/determined/_experiment_config.py (hunk 1 above)
Comment (rb-determined-ai) [reply harness/determined/_experiment_config.py:11]: more dead code, I'm pretty sure.  Actually, now that I look around it's totally unused, so I'll just remove it.

**Pull request merged**